homeassistant/components/template/trigger_entity.py

```diff
@@ -71,3 +84,93 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         self._process_data()
         self.async_write_ha_state()
+
+    def _render_single_template(
+        self,
+        key: str,
+        variables: dict[str, Any],
+        is_complex: bool,
+        strict: bool = False,
+    ) -> Any:
+        """Render a single template."""
+        try:
+            if is_complex:
+                return render_complex(self._config[key], variables)
+
+            return self._config[key].async_render(
+                variables, parse_result=key in self._parse_result, strict=strict
+            )
+        except TemplateError as err:
+            logging.getLogger(f"{__package__}.{self.entity_id.split('.')[0]}").error(
+                "Error rendering %s template for %s: %s",
+                key,
+                self.entity_id,
+                err,
+            )
+
+        return _SENTINEL
+
+    def _render_templates(self, variables: dict[str, Any]) -> None:
+        """Render templates."""
+        rendered = dict(self._static_rendered)
+
+        # Check availability first
+        available = True
+        if CONF_AVAILABILITY in self._to_render_simple:
+            if (
+                result := self._render_single_template(
+                    CONF_AVAILABILITY, variables, False, True
+                )
+            ) is not _SENTINEL:
+                rendered[CONF_AVAILABILITY] = available = result
+
+        if not available:
+            self._rendered = rendered
+            return
+
+        # If state fails to render, the entity should go unavailable.
+        if CONF_STATE in self._to_render_simple:
+            if (
+                result := self._render_single_template(CONF_STATE, variables, False)
+            ) is _SENTINEL:
+                self._rendered = self._static_rendered
+                return
+
+            rendered[CONF_STATE] = result
+
+        for key in self._to_render_simple:
+            # Skip availability because we already handled it before.
+            if key in (CONF_AVAILABILITY, CONF_STATE):
+                continue
+
+            if (
+                result := self._render_single_template(key, variables, False)
+            ) is not _SENTINEL:
+                rendered[key] = result
+
+        for key in self._to_render_complex:
+            if (
+                result := self._render_single_template(key, variables, True)
+            ) is not _SENTINEL:
+                rendered[key] = result
+
+        if CONF_ATTRIBUTES in self._config:
+            attributes = {}
+            for attribute, attribute_template in self._config[CONF_ATTRIBUTES].items():
```

Comment:
Do we really need to split this out and handle individual errors per attribute? Do we do that in the state based template entity?

Reply:
Yes, if we don't do that, any attribute error will cause all attributes to disappear.  This was split to ensure an error in a single attribute does not affect other attribute resolutions.

Reply:
OK, but is the new behavior aligned with state based template entities?

Reply:
Yep!  That's why I kept it as a loop.  The other option is to make a class that handles them separately but why bother when it reads better where it is.